### ISSUE TITLE
Brackets in brackets

### ISSUE THREAD

**Issue by Madrain** [TonyGravagno/MVBASIC-plus-plus#4]
brackets in brackets are not rendered correctly

![image](https://user-images.githubusercontent.com/30535720/28743012-ba95ada8-7447-11e7-8361-635a43a09ddc.png)

the last part should be red too

**Comment by TonyGravagno** [comment]
Yup, it's not smart enough to count open/close delimiters. This little experiment is showing its limitations. I think the only way to address these parsing/lexing issues is to create a "real" language definition.

I'm flagging some of these issues as Needs Research. They're technically not bugs in the code because they can't be fixed given the limitations of the technology. I just need to research alternatives.

BTW, for the example, the equivalent code is: `SBPM<-1> = "<div....." : ... : ...`
